msu/systems/mod_settings/mod_settings_mod_addon.nut

```diff
@@ -39,6 +39,8 @@
 
 	function requireSettingValue( _setting, _value )
 	{
+		::MSU.System.ModSettings.registerRequiredSettingValue(this.getMod(), _setting, _value);
+
 		if (_setting.getValue() != _value)
 		{
 			if (_setting.isLocked())
```

Comment:
Move this and the rest of the bottom of this function into a new function in ModSettingSystem called requireSettingValue, then when you call this from the backend during deserialization you only call that function and not this one which means you don't have to clone.

Reply:
Excellent idea.